lib/service/connect.go

```diff
@@ -1047,7 +1047,7 @@ func (process *TeleportProcess) rotate(conn *Connector, localState auth.StateV2,
 // For config v1 and v2, it will attempt to direct dial the auth server, and fallback to trying to tunnel
 // to the Auth Server through the proxy.
 func (process *TeleportProcess) newClient(identity *auth.Identity) (*auth.Client, error) {
-	if identity.ID.Role != types.RoleInstance {
+	if identity.ID.Role != types.RoleInstance && identity.ID.Role != types.RoleMDM {
```

Comment:
This is as quickly I could come up with but lmk if there's more better way to resolve this.

Reply:
FYI @codingllama @justinas tagging you both if there's anything else to consider from Jamf service and hosted plugin perspective.

Reply:
Definitely want @fspmarshall to look at this as well.

Reply:
Could someone spell out the consequences of this change?

Would this same condition hit other plugins, or future services that don't need extraordinary permissions?

Would it be correct for plugins to request RoleInstance as well, instead of this?

Reply:
This is harmless, it just causes allocation of mdm certs to fallback to the pre-https://github.com/gravitational/teleport/pull/30384 behavior.